data/data/ovirt/cluster/masters/main.tf

```diff
@@ -22,23 +31,71 @@ resource "ovirt_vm" "master" {
   # VM type server or high performance -> clone = true
   clone = var.ovirt_master_clone != null ? tobool(var.ovirt_master_clone) : (var.ovirt_master_vm_type == "desktop" ? false : true)
 
-  initialization {
-    host_name     = "${var.cluster_id}-master-${count.index}"
-    custom_script = var.ignition_master
-  }
+  # Initialization sets the host name and script run when the machine first starts.
+  initialization_host_name     = "${var.cluster_id}-master-${count.index}"
+  initialization_custom_script = var.ignition_master
 
-  block_device {
-    interface      = "virtio_scsi"
-    size           = var.ovirt_master_os_disk_size_gb
-    format         = var.ovirt_master_format != "" ? var.ovirt_master_format : null
-    sparse         = tobool(var.ovirt_master_sparse)
-    storage_domain = var.ovirt_storage_domain_id
+  # Placement policy dictates which hosts this master can run on.
+  #
+  # TODO there may be a bug here since we are pinning the masters to the existing detected hosts and this is never
+  #      updated.
+  placement_policy_affinity = var.ovirt_master_auto_pinning_policy != "" ? "migratable" : ""
+  placement_policy_host_ids = var.ovirt_master_auto_pinning_policy != "" ? data.ovirt_cluster_hosts.master.*.id : []
```

Comment:
this is a mistake, the correct way to get the host IDs  `data.ovirt_cluster_hosts.list.hosts.*.id`

Reply:
see an example here https://github.com/oVirt/terraform-provider-ovirt/pull/405/files#diff-d8edb2d925bab5644f323d25c5878b9dccf00d3f1fa6d4b7d3dc3a5cecb4443cR17